Docs/reference/content/reference/driver/crud/client_side_encryption.md

```diff
@@ -0,0 +1,264 @@
+            clientEncryption.Dispose();
+
+            var collectionNamespace = CollectionNamespace.FromFullName("test.coll");
+
+            var schemaMap = $@"{{
+                properties: {{
+                    encryptedField: {{
+                        encrypt: {{
+                            keyId: [{{
+                                '$binary' : {{
+                                    'base64' : '{base64DataKeyId}',
+                                    'subType' : '04'
+                                }}
+                            }}],
+                        bsonType: 'string',
+                        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+                        }}
+                    }}
+                }},
+                'bsonType': 'object'
+            }}";
+            var autoEncryptionSettings = new AutoEncryptionOptions(
+                keyVaultNamespace,
+                kmsProviders,
+                schemaMap: new Dictionary<string, BsonDocument>()
+                {
+                    { collectionNamespace.ToString(), BsonDocument.Parse(schemaMap) }
+                });
+            var clientSettings = new MongoClientSettings
+            {
+                AutoEncryptionOptions = autoEncryptionSettings
+            };
+            var client = new MongoClient(clientSettings);
+            var database = client.GetDatabase("test");
+            database.DropCollection("coll");
+            var collection = database.GetCollection<BsonDocument>("coll");
+
+            collection.InsertOne(new BsonDocument("encryptedField", "123456789"));
+
+            var result = collection.Find(FilterDefinition<BsonDocument>.Empty).First();
+            Console.WriteLine(result.ToJson());
+        }
+    }
+}
+```
+
+{{% note %}}
+Auto encryption is an **enterprise** only feature.
+{{% /note %}}
+
+The following example shows how to configure the `AutoEncryptionSettings`
+instance to create a new key and setting the json schema map. The following
+example has been adapted from
+[`ClientSideEncryptionExamples.cs`](https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs), which can be found on Github along with the driver source. 
+
+```csharp
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using MongoDB.Bson;
+using MongoDB.Driver.Encryption;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MongoDB.Driver.Examples
+{
+    public class ClientEncryptionExamples
+    {
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+        public void ClientSideEncryptionAutoEncryptionSettingsTour()
```

Comment:
Isn't this example the same as in line `104`? Do we need to have two the same examples?

Reply:
Ah good catch, the examples were meant to be different. Fixed!